automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala

```diff
@@ -0,0 +1,117 @@
+import DefaultJsonProtocol._
+import org.scalatest.time.{Minutes, Seconds, Span}
+import org.scalatest.concurrent.Eventually
+
+
+class MethodLaunchSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with Matchers
+  with CleanUp with RandomUtil with Retry with Eventually
+  with BillingFixtures with WorkspaceFixtures with MethodFixtures {
+
+  val methodConfigName: String = SimpleMethodConfig.configName + "_" + UUID.randomUUID().toString
+  val operations = Array(Map("op" -> "AddUpdateAttribute", "attributeName" -> "participant1", "addUpdateAttribute" -> "testparticipant"))
+  val entity: Array[Map[String, Any]] = Array(Map("name" -> "participant1", "entityType" -> "participant", "operations" -> operations))
+
+  "launch workflow with input not defined" in {
+    val user = UserPool.chooseProjectOwner
+    implicit val authToken: AuthToken = user.makeAuthToken()
+    withCleanBillingProject(user) { billingProject =>
+      withWorkspace(billingProject, "MethodLaunchSpec_launch_workflow_input_not_defined") { workspaceName =>
+        Rawls.entities.importMetaData(billingProject, workspaceName, entity)
+
+        withMethod("MethodLaunchSpec_input_undefined", MethodData.InputRequiredMethod, 1) { methodName =>
+          val method = MethodData.InputRequiredMethod.copy(methodName = methodName)
+          Rawls.methodConfigs.createMethodConfigInWorkspace(billingProject, workspaceName, method,
+            method.methodNamespace, methodConfigName, 1, Map.empty, Map.empty, method.rootEntityType)
+          val exception = intercept[RestException](Rawls.submissions.launchWorkflow(billingProject, workspaceName, method.methodNamespace, methodConfigName, "participant",
+          "participant1", "this", false))
+          exception.message.parseJson.asJsObject.fields("message").convertTo[String].contains("Missing inputs:") shouldBe true
+        }
+      }
+    }
+  }
+
+
+  "owner can abort a launched submission" in {
+    val user = UserPool.chooseProjectOwner
+    implicit val authToken: AuthToken = user.makeAuthToken()
+    withCleanBillingProject(user) { billingProject =>
+      withWorkspace(billingProject, "MethodLaunchSpec_abort_submission") { workspaceName =>
+
+        Rawls.entities.importMetaData(billingProject, workspaceName, entity)
+
+        withMethod("MethodLaunchSpec_abort", MethodData.SimpleMethod) { methodName =>
+          val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+          Rawls.methodConfigs.createMethodConfigInWorkspace(billingProject, workspaceName,
+            method, method.methodNamespace, method.methodName, 1,
+            SimpleMethodConfig.inputs, SimpleMethodConfig.outputs, method.rootEntityType)
+
+          val submissionId = Rawls.submissions.launchWorkflow(billingProject, workspaceName, method.methodNamespace, method.methodName, method.rootEntityType, "participant1", "this", false)
+
+          Rawls.submissions.abortSubmission(billingProject, workspaceName, submissionId)
```

Comment:
I would check submission status first before attempt to abort. Sometimes, new submission status is `Error` instead `Submitted`, test should just fail here, not continued.

Reply:
good point!